sdk/nodejs/iot/topicRule.ts

```diff
@@ -164,25 +164,25 @@ export class TopicRule extends pulumi.CustomResource {
         if (opts.id) {
             const state = argsOrState as TopicRuleState | undefined;
             resourceInputs["arn"] = state ? state.arn : undefined;
-            resourceInputs["cloudwatchAlarm"] = state ? state.cloudwatchAlarm : undefined;
+            resourceInputs["cloudwatchAlarms"] = state ? state.cloudwatchAlarms : undefined;
             resourceInputs["cloudwatchLogs"] = state ? state.cloudwatchLogs : undefined;
-            resourceInputs["cloudwatchMetric"] = state ? state.cloudwatchMetric : undefined;
+            resourceInputs["cloudwatchMetrics"] = state ? state.cloudwatchMetrics : undefined;
             resourceInputs["description"] = state ? state.description : undefined;
-            resourceInputs["dynamodb"] = state ? state.dynamodb : undefined;
+            resourceInputs["dynamodbs"] = state ? state.dynamodbs : undefined;
             resourceInputs["dynamodbv2s"] = state ? state.dynamodbv2s : undefined;
-            resourceInputs["elasticsearch"] = state ? state.elasticsearch : undefined;
+            resourceInputs["elasticsearches"] = state ? state.elasticsearches : undefined;
             resourceInputs["enabled"] = state ? state.enabled : undefined;
             resourceInputs["errorAction"] = state ? state.errorAction : undefined;
-            resourceInputs["firehose"] = state ? state.firehose : undefined;
+            resourceInputs["firehoses"] = state ? state.firehoses : undefined;
             resourceInputs["https"] = state ? state.https : undefined;
             resourceInputs["iotAnalytics"] = state ? state.iotAnalytics : undefined;
             resourceInputs["iotEvents"] = state ? state.iotEvents : undefined;
             resourceInputs["kafkas"] = state ? state.kafkas : undefined;
-            resourceInputs["kinesis"] = state ? state.kinesis : undefined;
-            resourceInputs["lambda"] = state ? state.lambda : undefined;
+            resourceInputs["kineses"] = state ? state.kineses : undefined;
+            resourceInputs["lambdas"] = state ? state.lambdas : undefined;
             resourceInputs["name"] = state ? state.name : undefined;
-            resourceInputs["republish"] = state ? state.republish : undefined;
-            resourceInputs["s3"] = state ? state.s3 : undefined;
+            resourceInputs["republishes"] = state ? state.republishes : undefined;
+            resourceInputs["s3Buckets"] = state ? state.s3Buckets : undefined;
```

Comment:
Is it intentional that s3 became s3Buckets?

Reply:
Yes. I was trying to come up with something that had an understandable plural and the associated object seemed bucket specific (`bucketName` was a field). If we don't have something more specific then that, I can just pin `s3`.

Reply:
Fixed (moved back to `s3` with sub-field `s3` in https://github.com/pulumi/pulumi-aws/pull/2615/commits/28e1921c4b75c6e88db8281312d4b817ef3e7825.